wagtail/core/actions/copy_for_translation.py

```diff
@@ -0,0 +1,152 @@
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+
+from wagtail.core.utils import find_available_slug
+
+
+class ParentNotTranslatedError(Exception):
+    """
+    Raised when a call to Page.copy_for_translation is made but the
+    parent page is not translated and copy_parents is False.
+    """
+
+    pass
+
+
+class CopyForTranslationPermissionError(PermissionDenied):
+    """
+    Raised when the page translation copy cannot be performed due to insufficient permissions.
+    """
+
+    pass
+
+
+class CopyForTranslationAction:
```

Comment:
Perhaps we should call this ``CopyPageForTranslationAction`` as we include ``Page`` in the other action classes